src/Tools/dotnet-monitor/Experimental/ExperimentalFeatures.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.Options;
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Diagnostics.Tools.Monitor
+{
+    internal sealed class ExperimentalFeatures : IExperimentalFeatures
+    {
+        private readonly InProcessFeaturesOptions _options;
+
+        public ExperimentalFeatures(IOptions<InProcessFeaturesOptions> options)
+        {
+            _options = options.Value;
+        }
+
+        public bool IsCallStacksEnabled => _options.Enabled && ExperimentalFlags.IsCallStacksEnabled;
```

Comment:
Do you want to make this class more generic? For example, once stacks is not an experimental feature but inproc features are disabled, we'll still need to test the features much like we do now, but just not the experimental flag.